helm/action/helper_test.go

```diff
@@ -0,0 +1,36 @@
+package action
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+var helmRoot string
+
+func init() {
+	helmRoot = filepath.Join(os.Getenv("GOPATH"), "src/github.com/deis/helm/")
+}
+
+func createTmpHome() string {
+	// create a temp directory
+	tmpHome, _ := ioutil.TempDir("", "helm_home")
+	defer os.Remove(tmpHome)
+
+	// get properties of tmpHome
+	dstat, _ := os.Stat(tmpHome)
+
+	// create cache directory
+	tmpHomeCache := filepath.Join(tmpHome, "cache")
+	os.Mkdir(tmpHomeCache, dstat.Mode())
+
+	// absolute path to testdata charts
+	testChartsPath := filepath.Join(helmRoot, "helm/testdata/charts")
+
+	// copy testdata charts into cache
+	// mock git clone
+	exec.Command("cp", "-R", testChartsPath, tmpHomeCache).Run()
```

Comment:
There's a copyDir() function in (I think) `action/update.go` that does essentially this.